### ISSUE TITLE
Examples does not compile on aarch64: g++: error: unrecognized command-line option '-m64'

### ISSUE THREAD

**Issue by vt-alt** [ispc/ispc#2068]
```
builder@aarch64:/tmp/build$ cmake /usr/share/doc/ispc-1.15.0/examples
-- The C compiler identification is GNU 10.2.1
-- The CXX compiler identification is GNU 10.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/build
builder@aarch64:/tmp/build$ make VERBOSE=1
/usr/bin/cmake -S/usr/share/doc/ispc-1.15.0/examples -B/tmp/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /tmp/build/CMakeFiles /tmp/build//CMakeFiles/progress.marks
make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/tmp/build'
make  -f aobench/CMakeFiles/aobench.dir/build.make aobench/CMakeFiles/aobench.dir/depend
make[2]: Entering directory '/tmp/build'
[  1%] Generating ao_ispc.h, ao_ispc.o
cd /tmp/build/aobench && /usr/bin/ispc /usr/share/doc/ispc-1.15.0/examples/aobench/ao.ispc --pic --target=neon --arch=aarch64 -h /tmp/build/aobench/ao_ispc.h -o /tmp/build/aobench/ao_ispc.o
cd /tmp/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /usr/share/doc/ispc-1.15.0/examples /usr/share/doc/ispc-1.15.0/examples/aobench /tmp/build /tmp/build/aobench /tmp/build/aobench/CMakeFiles/aobench.dir/DependInfo.cmake --color=
Dependee "/tmp/build/aobench/CMakeFiles/aobench.dir/DependInfo.cmake" is newer than depender "/tmp/build/aobench/CMakeFiles/aobench.dir/depend.internal".
Dependee "/tmp/build/aobench/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/tmp/build/aobench/CMakeFiles/aobench.dir/depend.internal".
Scanning dependencies of target aobench
make[2]: Leaving directory '/tmp/build'
make  -f aobench/CMakeFiles/aobench.dir/build.make aobench/CMakeFiles/aobench.dir/build
make[2]: Entering directory '/tmp/build'
[  2%] Building CXX object aobench/CMakeFiles/aobench.dir/ao.cpp.o
cd /tmp/build/aobench && /usr/bin/c++  -I/tmp/build/aobench -m64 -fpie -O2 -o CMakeFiles/aobench.dir/ao.cpp.o -c /usr/share/doc/ispc-1.15.0/examples/aobench/ao.cpp
aarch64-alt-linux-g++: error: unrecognized command-line option '-m64'
make[2]: *** [aobench/CMakeFiles/aobench.dir/build.make:90: aobench/CMakeFiles/aobench.dir/ao.cpp.o] Error 1
make[2]: Leaving directory '/tmp/build'
make[1]: *** [CMakeFiles/Makefile2:407: aobench/CMakeFiles/aobench.dir/all] Error 2
make[1]: Leaving directory '/tmp/build'
make: *** [Makefile:149: all] Error 2
```
Without `-m64` they compile ok.

**Comment by dbabokin** [comment]
What OS is it? What hardware is it?

**Comment by vt-alt** [comment]
ALT Linux on aarch64. It seems that gcc-c++ does not support -m64/-m32 (unlike clang) while cmake config 'thinks' it does.
```
    if (UNIX)
        if (${ARCH_BIT} EQUAL 32)
            target_compile_options(${example_NAME} PRIVATE -m32)
        else()
            target_compile_options(${example_NAME} PRIVATE -m64)
        endif()
```

**Comment by dbabokin** [comment]
This was fixed since `v1.15.0`. You can try `main` branch to verify that. `aarch64` is currently built and tested on every commit in [TravisCI](https://travis-ci.com/github/ispc/ispc). Please reopen if there are problems with `main`.
